Add support for ALL grant on mysql >= 8.0.33

alaunay opened the pull request:
## Summary

This patch add ALL grant support for mysql >= 8.0.33, due tot the recent addition of the new TELEMETRY_LOG_ADMIN, see:
https://docs.oracle.com/cd/E17952_01/mysql-8.0-relnotes-en/news-8-0-33.html#mysqld-8-0-33-feature

## Additional Context
Add any additional context about the problem here. 
- [x] Root cause and the steps to reproduce. (If applicable)

Puppet changes grant at each run:

```
Notice: /Stage[main]/Mysql::Server::Providers/Mysql_grant[XXX@YYY/*.*]/privileges: privileges changed ['ALL', 'APPLICATION_PASSWORD_ADMIN', 'AUDIT_ABORT_EXEMPT', 'AUDIT_ADMIN', 'AUTHENTICATION_POLICY_ADMIN', 'BACKUP_ADMIN', 'BINLOG_ADMIN', 'BINLOG_ENCRYPTION_ADMIN', 'CLONE_ADMIN', 'CONNECTION_ADMIN', 'ENCRYPTION_KEY_ADMIN', 'FIREWALL_EXEMPT', 'FLUSH_OPTIMIZER_COSTS', 'FLUSH_STATUS', 'FLUSH_TABLES', 'FLUSH_USER_RESOURCES', 'GROUP_REPLICATION_ADMIN', 'GROUP_REPLICATION_STREAM', 'INNODB_REDO_LOG_ARCHIVE', 'INNODB_REDO_LOG_ENABLE', 'PASSWORDLESS_USER_ADMIN', 'PERSIST_RO_VARIABLES_ADMIN', 'REPLICATION_APPLIER', 'REPLICATION_SLAVE_ADMIN', 'RESOURCE_GROUP_ADMIN', 'RESOURCE_GROUP_USER', 'ROLE_ADMIN', 'SENSITIVE_VARIABLES_OBSERVER', 'SERVICE_CONNECTION_ADMIN', 'SESSION_VARIABLES_ADMIN', 'SET_USER_ID', 'SHOW_ROUTINE', 'SYSTEM_USER', 'SYSTEM_VARIABLES_ADMIN', 'TABLE_ENCRYPTION_ADMIN', 'TELEMETRY_LOG_ADMIN', 'XA_RECOVER_ADMIN'] to ['ALL'] (corrective)
```

- [x] Thought process behind the implementation.

Add the new TELEMETRY_LOG_ADMIN, according to previous code.

## Related Issues (if any)

Issue #1474 #1502 #1503

## Checklist
- [ ] 🟢 Spec tests.
- [ ] 🟢 Acceptance tests.
- [X] Manually verified. (For example `puppet apply`)

BuJo committed:
Fix backup/rotation with multiple excluded databases

* When using multiple excluded databases, the list of databases
  is filtered using `grep -v`.
  i.e. `grep -v '^\(information_schema|performance_schema\)$`
* When using Basic vs Extended Regular Expressions, the characters
  `(` and `|` lose their special meaning, the backslashed versions
  have to be used.
  For the group (`()`) the escaping has been done, however the
  alternation is unescaped.

Leading to:

* All the excluded databases will be backed up.
* In case a database is not backuppable (which is why it had been
  excluded), this leads to the cleanup not being run at all, as it
  depends on the backup having been successful.

This MR aims to fix this issue, by revising the regular expression
and specifying that behaviour in the respective class spec.

alaunay committed:
Add support for ALL grant on mysql >= 8.0.33 (add the new TELEMETRY_LOG_ADMIN privilege)
https://docs.oracle.com/cd/E17952_01/mysql-8.0-relnotes-en/news-8-0-33.html#mysqld-8-0-33-feature

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/puppetlabs/puppetlabs-mysql?pullRequest=1618) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/puppetlabs/puppetlabs-mysql?pullRequest=1618) before we can accept your contribution.<br/>**3** out of **4** committers have signed the CLA.<br/><br/>:white_check_mark: alaunay<br/>:white_check_mark: Marc-DRI<br/>:white_check_mark: BuJo<br/>:x: rahuls2997<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/puppetlabs/puppetlabs-mysql?pullRequest=1618) it.</sub>

fraenki commented:
So much looking forward to this! (FWIW, this is also causing issues when performing long-running schema changes.)

@Ramesh7, maybe if you find some time, could you have a look at this please? Thanks :)

MarcWort commented:
Any progress in reviewing that PR?

fraenki commented:
ping @puppetlabs

Marc-DRI committed:
Fix spec unit for mysql_server_id

Value of the facter :macaddress was nil, causing tests to have two
failures (contexts "igalic's laptop" and "node with lo only").

Update the facter initialization allows to pass the tests
successfully.

ic248 commented:
Would be good to have an eta for this. Although can easily ignore it's a pain when developing and seeing constant changes

alaunay committed:
Merge branch 'main' into add_support_for_mysql_greater_than_8_0_33

rahuls2997 committed:
Fix mysql_users failing for SLES

fix mistakes

fix spec for the changed query

ianc769 commented:
Maybe @alexjfisher or @bastelfreak ? Been a few months 😅

bastelfreak commented:
@ianc769 can you please rebase against main to get rid of the merge commit?

bastelfreak commented on lib/puppet/provider/mysql_grant/mysql.rb:
```diff
@@ -91,8 +91,16 @@ def self.instances
+
         # rubocop:disable Layout/LineLength
-        if (newer_than('mysql' => '8.0.0') && (sorted_privileges == mysql_v8_privileges || sorted_privileges == mysqlcluster_v8_privileges)) || sorted_privileges == mysql_pre_v8_privileges_one || sorted_privileges == mysql_pre_v8_privileges_two
+        if (newer_than('mysql' => '8.0.0') && (sorted_privileges == mysql_v8_privileges || sorted_privileges == mysqlcluster_v8_privileges)) || sorted_privileges == mysql_pre_v8_privileges_one || sorted_privileges == mysql_pre_v8_privileges_two || sorted_privileges == mysql_pre_v8_privileges_three
```
can we somehow have an acceptance tests for this?

amitkarsale replied:
@alaunay as @bastelfreak mentioned can you please add acceptance test, Thanks!

alaunay committed:
Merge branch 'main' into add_support_for_mysql_greater_than_8_0_33

ianc769 commented:
> [![CLA assistant check](https://camo.githubusercontent.com/bb14f9741a98be03eb447ca21a4a5e96201df401e5be70f9ef52130f4e1ff903/68747470733a2f2f636c612d617373697374616e742e696f2f70756c6c2f62616467652f6e6f745f7369676e6564)](https://cla-assistant.io/puppetlabs/puppetlabs-mysql?pullRequest=1618) Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/puppetlabs/puppetlabs-mysql?pullRequest=1618) before we can accept your contribution.**3** out of **4** committers have signed the CLA.✅ alaunay✅ Marc-DRI✅ BuJo❌ rahuls2997You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/puppetlabs/puppetlabs-mysql?pullRequest=1618) it.

Is the cla, required to progress?

bastelfreak commented:
Given how perforce treats the community since half a year, I recommend to not sign the CLA (https://www.puppet.com/blog/open-source-puppet-updates-2025).

alaunay committed:
Merge branch 'puppetlabs:main' into add_support_for_mysql_greater_than_8_0_33

jst-cyr commented:
@alaunay : It seems like quite a few people were interested in this one. I'm reviewing PRs and came across this one. Is there still interest in adding this feature?

alaunay commented:
Hey Jason @jst-cyr , nice to see you out of IRC/slack ;-) I think there is interest. We forked the module from our side to support it, and recently upgraded the patch to work with mysql 8.4. Right now though, I'm not really inclined to patch again here: a 2 and a half year answer is kinda a deterrent. Available on IRC if you want to discuss it more at length, but I suspect you already know the feeling :-)

jst-cyr commented:
Completely fair, @alaunay , and I am sorry it took so long. I'm going to leave this PR open then and if it gets picked up we'll take a look at the changes here and perhaps open a newer version that is patched up.
